examples/client/src/MessageSender.cpp

```diff
@@ -0,0 +1,175 @@
+// Concord
+//
+// Copyright (c) 2018-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "MessageSender.hpp"
+#include "KVMessageConsumer.hpp"
+#include "KVMessageGenerator.hpp"
+
+using namespace concord::osexample;
+
+std::unique_ptr<IMessageGenerator> MessageSender::getMessageGenerator(uint16_t executionEngineType) {
+  switch (executionEngineType) {
+    case ExecutionEngineType::KV:
```

Comment:
This may change later as we may want to guess the execution engine type from the received message or from a config.